retworkx-core/src/shortest_path/dijkstra.rs

```diff
@@ -169,5 +170,11 @@ where
         }
         visited.visit(node);
     }
-    Ok(scores)
+
+    Ok(graph
+        .node_identifiers()
+        .filter_map(|node_id| {
+            scores[graph.to_index(node_id)].map(|score| (node_id, score))
+        })
+        .collect())
```

Comment:
Iterating directly over scores should be faster.
```suggestion
    Ok(scores
        .into_iter()
        .enumerate()
        .filter_map(|(node, score)| {
            score.map(|val| (graph.from_index(node), val))
        })
        .collect())
```